Storage/src/StorageObject.php

```diff
@@ -672,13 +683,67 @@ public function downloadToFile($path, array $options = [])
      */
     public function downloadAsStream(array $options = [])
```

Comment:
Snippet test for this appears to be failing: `StorageObjectTest::testDownloadAsStreamWithRangeHeaders`